packages/amplify-prompts/src/progressbars/spinner.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * This class is created with the hope that one day we will move awy from ora
+ * and use a re writable block instead.
+ */
+import chalk from 'chalk';
+import { AmplifyTerminal as Terminal } from './terminal';
+
+/**
+ * Amplify spinner instance
+ */
+export class AmplifySpinner {
+    private frameCount : number;
+    private frames : string[];
+    private timer!: ReturnType<typeof setTimeout>;
+    private prefixText : string;
+    private terminal: Terminal;
+    private refreshRate: number;
+
+    constructor(text : string) {
+      this.frameCount = 0;
+      this.frames = ['⠋', '⠙', '⠹', '⠸', '⠼', '⠴', '⠦', '⠧', '⠇', '⠏'];
+      this.prefixText = text;
+      this.terminal = new Terminal();
+      this.refreshRate = 300;
+    }
+
+    /**
+     * Render function
+     */
+    render() : void {
+      if (this.timer) {
+        clearTimeout(this.timer);
+      }
+      const lines = [`${this.prefixText} ${chalk.blue(this.frames[this.frameCount])}`];
+      this.frameCount = this.frames.length < this.frameCount ? 0 : this.frameCount + 1;
+      this.terminal.writeLines(lines);
+      this.timer = setTimeout(this.render.bind(this), this.refreshRate);
```

Comment:
I _think_ if you change this to
```suggestion
      this.timer = setTimeout(() => this.render(), this.refreshRate);
```
then we don't need to worry about this binding. Not a huge deal but in general if we can avoid it we should because it can lead to confusion down the road